client/src/app/plan.service.ts

```diff
@@ -98,6 +114,93 @@ export class PlanService {
     return this.user.pipe(map(mapCallback));
   }
 
+  /**
+   * Formats the courses in the program into an object that the component can read and render:
+   * courseList: Array<{
+   *  department: string;
+   *  courses: Array<{
+   *    department: string;
+   *    code: string;
+   *    school: string;
+   *    title: string;
+   *    description: string;
+   *    credit: string;
+   *    difficulty: number;
+   *    impaction: number;
+   *  }>;
+   * }>
+   * @param program The degree program object
+   */
+  getCourseList(program: any): Array<any> {
```

Comment:
Converting the requirements field of the program into a format that the component can render better.